x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/aggregate/StdDeviation.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.expression.function.aggregate;
+
+import org.elasticsearch.common.io.stream.NamedWriteableRegistry;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.compute.aggregation.AggregatorFunctionSupplier;
+import org.elasticsearch.compute.aggregation.StdDeviationDoubleAggregatorFunctionSupplier;
+import org.elasticsearch.compute.aggregation.StdDeviationIntAggregatorFunctionSupplier;
+import org.elasticsearch.compute.aggregation.StdDeviationLongAggregatorFunctionSupplier;
+import org.elasticsearch.xpack.esql.EsqlIllegalArgumentException;
+import org.elasticsearch.xpack.esql.core.expression.Expression;
+import org.elasticsearch.xpack.esql.core.expression.Literal;
+import org.elasticsearch.xpack.esql.core.tree.NodeInfo;
+import org.elasticsearch.xpack.esql.core.tree.Source;
+import org.elasticsearch.xpack.esql.core.type.DataType;
+import org.elasticsearch.xpack.esql.expression.function.Example;
+import org.elasticsearch.xpack.esql.expression.function.FunctionInfo;
+import org.elasticsearch.xpack.esql.expression.function.Param;
+import org.elasticsearch.xpack.esql.planner.ToAggregator;
+
+import java.io.IOException;
+import java.util.List;
+
+import static java.util.Collections.emptyList;
+
+public class StdDeviation extends AggregateFunction implements ToAggregator {
```

Comment:
I wasn't sure what best to name this. There were a few options: `StdDeviation`, `StandardDeviation`, or `Stdev` (or maybe even `Stddev`)

Reply:
What about for the name of the ES|QL function? Right now it's `std_deviation`, I feel like `std_dev` maybe works better? I worry that `standard_deviation` is kind of long

Reply:
I think `std_dev` is better.

Reply:
Should the internal name be changed to StdDev to match?

Reply:
I prefer `StandardDeviation` for the class name, but feel free to choose whichever name you prefer.

Reply:
I think it might be easier to change both; I tried changing only the function name but it looks like the name of the tests class is used for generating the docs so I think it might be simpler to use StdDev for the class name as well